### DISCUSSION TITLE
How can I dynamically create a model and database table ?

### DISCUSSION THREAD

**Discussion by gemufeng** [yahoo/elide#2619] [General]
I want to client can create a database table,pass field name and type from restapi, and then the back-end can automatically generate model, database table and API. 
How can I do it?

**Comment by aklish** [comment]
Some parts of Elide are not friendly for auto-generating models.  ORMs like Hibernate do byte code manipulation of JVM classes - so they require the model to be recompiled.

That said, you could create a custom DataStore to manage your dynamic models (similar to what Elide does for Analytic models).  Analytic models can be dynamically created and registered for example.  

I would need more details on your use case.

**Comment by gemufeng** [comment]
> 
I am developing a CMS system, which allows users to create dynamic model then generate restfull API,elide fully meets the functions of restful API and graphql API, but I don't know how to dynamically generate model and table.

**Reply by chirdeeptomar** [reply]
If you are open to systems outside Java, check out https://strapi.io/ which is a CMS written in node and does exactly what you are after and exposes both REST and GraphQL.

**Comment by aklish** [comment]
This would be outside Elide - but what about a workflow that takes user input, feeds that to a templating engine to generate code/models, build the models, and redeploys your service?

**Reply by thaingo** [reply]
> This would be outside Elide - but what about a workflow that takes user input, feeds that to a templating engine to generate code/models, build the models, and redeploys your service?

Which templating engine you recommend for this purpose @aklish ?

**Comment by aklish** [comment]
Another great one is liquid.

**Comment by aklish** [comment]
I like handlebars but I’m sure there are other great options.

**Comment by chirdeeptomar** [comment]
Apache Velocity.
